File: test/resources/repos/csharp/test_repo/BaseClass.cs
using System;

namespace TestRepo
{
    public abstract class BaseClass
    {
        protected string name;

        public BaseClass(string name)
        {
            this.name = name;
        }

        public abstract void Execute();

        public string GetName()
        {
            return name;
        }
    }

    public interface IProcessable
    {
        void Process();
        string GetType();
    }

    public interface IReadable
    {
        string Read();
    }

    public interface IWritable
    {
        void Write(string data);
    }
}

File: test/resources/repos/typescript/test_repo/BaseClass.ts
export abstract class BaseClass {
    protected name: string;

    constructor(name: string) {
        this.name = name;
    }

    abstract execute(): void;

    getName(): string {
        return this.name;
    }
}

export interface Processable {
    process(): void;
    getType(): string;
}

export interface Readable {
    read(): string;
}

export interface Writable {
    write(data: string): void;
}